include/memory/memory.hh

```diff
@@ -32,17 +35,50 @@ public:
   Word loadWord(Addr addr);
   void storeWord(Addr addr, Word word);
 
+  void printMemStats(std::ostream &ost) const;
+  const MemoryStats &getMemStats() const;
+
   template <std::forward_iterator It>
   void storeRange(Addr start, It begin, It end) {
     std::for_each(begin, end, [&start, this](auto curWord) {
       storeWord(start, curWord);
       start += kXLENInBytes;
     });
   }
+};
 
-  std::size_t getCurrMemSize() const;
-  void printMemStats(std::ostream &ost) const;
-  const MemoryStats &getMemStats() const;
+class PhysMemory final {
+public:
+  struct Page {
+    Page() = default;
+
+    std::array<Word, kPageSize / sizeof(Word)> wordStorage{};
```

Comment:
I suppose you should replace `std::array` w/ `std::vector` here. At least because of possible stack overflow here

Reply:
Done